### ISSUE TITLE
Using native eslint rules config in .eslintrc

### ISSUE THREAD

**Issue by avalanche1** [typescript-eslint/typescript-eslint#823]
`typescript-eslint` is reporting errors on rules I have disabled in `.eslintrc`. 
How can I use basic eslint rules instead of typing `rules: {  @typescript-eslint/camelcase: "off" }`?

**What code were you trying to parse?**

`Code:`
```ts
function get_json(){
  ///some code
}
```

`.eslintrc.yml:`
```yaml
env:
  node: true
  browser: true
  jest/globals: true
extends:
  - "eslint:recommended"
  - "plugin:@typescript-eslint/recommended"
  - "plugin:@typescript-eslint/eslint-recommended"
  - "hardcore-fp"
  - "plugin:jest/recommended"
  - "plugin:react/recommended"
  - "plugin:promise/recommended"
  - prettier
  - "prettier/@typescript-eslint"
parser: "@typescript-eslint/parser"
parserOptions:
  project: "./tsconfig.json"
  ecmaVersion: 2017,
  sourceType: module
plugins:
  - "@typescript-eslint/eslint-plugin"
  - jest
  - react
  - react-hooks
  - promise
overrides:
  - files:
      - "*.test.js"
      - "*.spec.js"
    rules:
      fp/no-unused-expression: "off"
rules:
  arrow-body-style: "off"
  camelcase: "off"
```

**What did you expect to happen?**
No errors reported.
**What actually happened?**
` error    Identifier 'get_json' is not in camel case           @typescript-eslint/camelcase`
**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `^1.13.0` |
| `TypeScript`                | `^3.5.3` |
| `ESLint`                    | `^5.16.0` |

**Comment by bradzacher** [comment]
The two rules are considered distinct - i.e. `@typescript-eslint/camelcase` and `camelcase` are two completely separate rules.

Why do you want to use the base rule from eslint? This rule has known issues with typescript code (hence we have created an extension rule within the plugin).

You would be better off configuring `@typescript-eslint/camelcase` with the setting you want.

---

Closing this for housekeeping purposes because it's not an open bug. Please continue commenting here.

**Comment by avalanche1** [comment]
Hey Brad, thanks for the answer.
Well, I wanted a kinda seamless experience when transitioning from js to ts. So I thought my old js `.eslintrc` would work out of the box; at least rule-wise..

**Comment by avalanche1** [comment]
And since I have cerca 30 manually specified rules - do I have to rename them all to `@typescript-eslint/{rule}`?

**Comment by JamesHenry** [comment]
> So I thought my old js .eslintrc would work out of the box; at least rule-wise..

This is documented on the main README, which everyone is strongly encouraged to read:

https://github.com/typescript-eslint/typescript-eslint#can-i-use-all-of-the-existing-eslint-plugins-and-rules-without-any-changes

**Comment by bradzacher** [comment]
Expanding on what James and the docs mentioned.

This is a limitation of eslint that stems from the fact that we're working in a pretty niche area of a superset language that uses eslint.

Unfortunately, in order to support new typescript features, we have to create a brand new rule within this plugin. Even if we build our rule off of the existing rule (eg below - our camelcase imports and builds on top of the base rule), eslint will always see the two rules as two separate entities. There is unfortunately no way for us to consume the config for the base rule and make this just work.

https://github.com/typescript-eslint/typescript-eslint/blob/42b3013ab846669fd730628f5cb0b043cfedabba/packages/eslint-plugin/src/rules/camelcase.ts#L5

You're also in a niche area for our "new user" userbase in that you're migrating a reasonably heavily customised config. For the most part people can just add our recommended config and be done with it. For you it'll take a few minutes of checking to get your config working.

The list of base rules that we have improved is here: https://github.com/typescript-eslint/typescript-eslint/blob/42b3013ab846669fd730628f5cb0b043cfedabba/packages/eslint-plugin/tools/generate-configs.ts#L22-L31

So all you have to do is check your 30 manually specified rules against this list, and prefix your config appropriately.

Also I see that you're using the `hardcore-fp` config, which depends on `eslint-config-hardcore`. So you'll want to double check that config's list, and override appropriately in your local config: https://github.com/EvgenyOrekhov/eslint-config-hardcore/blob/master/index.json

Other than that, everything else should "just work".

**Comment by avalanche1** [comment]
> This is documented on the main README, which everyone is strongly encouraged to read
https://github.com/typescript-eslint/typescript-eslint#can-i-use-all-of-the-existing-eslint-plugins-and-rules-without-any-changes

I'm sure I've read the readme. The thing is that particular section is rather abstract so at the time of reading I could not _infer_ the connection bw the readme and the issue at hand 🤕
